enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,349 @@
+results of the validation to derive defaults for the machine network
+CIDR, the API VIP, and other network configuration settings for the
+hosts.
+
+When enough hosts are configured, the assisted installer application
+replies to the agent on each host with the instructions it needs to
+take part in forming the cluster. The assisted installer application
+selects one host to run the bootstrap services used during
+installation, and the other hosts are told to write an RHCOS image to
+disk and set up ignition to fetch configuration from the
+machine-config-operator in the usual way.
+
+During installation, progress and error information is reported to the
+assisted installer application on `cloud.redhat.com` so it can be
+shown in the UI.
+
+### Disconnected Deployments
+
+The disconnected process is similar to the connected process, except
+that the assisted installer application must run somewhere in the
+disconnected environment, and the agents need to be directed there
+instead of to the cloud. There are several open questions related to
+how this would work.
+
+### Integration with Existing Bare Metal IPI Tools
+
+Clusters built using the assisted installer workflow use the same
+"baremetal" platform setting as clusters built with the IPI
+installer. The cluster runs metal3, without PXE booting support.
+
+BareMetalHosts created by the assisted installer workflow do not have
+BMC credentials set. This means that power-based fencing is not
+available for the associated nodes until the user provides the BMC
+details.
+
+### User Stories
+
+#### Story 1
+
+As a cluster deployer, I want to install OpenShift on a small set of
+hosts without having to make configuration changes to my network or
+obtain administrator access to infrastructure so I can experiment
+before committing to a full production-quality setup.
+
+#### Story 2
+
+As a cluster deployer, I want to install OpenShift on a large number
+of hosts using my existing provisioning tools to automate launching
+the installer so I can adapt my existing admin processes and
+infrastructure tools instead of replacing them.
+
+#### Story 3
+
+As a cluster deployer, I want to install a production-ready OpenShift
+cluster without committing to delegating all infrastructure control to
+the installer or to the cluster, so I can adapt my existing admin
+processes and infrastructure management tools instead of replacing
+them.
+
+#### Story 4
+
+As a cluster hardware administrator, I want to enable power control
+for the hosts that make up my running cluster so I can use features
+like fencing and failure remediation.
+
+### Implementation Details/Notes/Constraints
+
+Much of the work described by this enhancement already exists as a
+proof-of-concept implementation. Some aspects will need to change as
+part of moving from PoC to product. At the very least, the code will
+need to be moved into more a suitable GitHub org.
+
+The agent discussed in this design is different from the
```

Comment:
also the state machine, and the mode it operates

Reply:
I wanted to emphasize that there is a new component (the new agent). The state machine is part of the web app, right? I think for this document we can probably leave out that detail. Let me know if you disagree.